example/fine_advection/stokes.hpp

```diff
@@ -0,0 +1,150 @@
+                           Real w1, Real w2, MeshData<Real> *out) {
+  auto pack1 = pd.GetPack(in1);
+  auto pack2 = pd.GetPack(in2);
+  auto pack_out = pd.GetPack(out);
+
+  IndexRange ib = in1->GetBoundsI(cl, IndexDomain::entire, te);
+  IndexRange jb = in1->GetBoundsJ(cl, IndexDomain::entire, te);
+  IndexRange kb = in1->GetBoundsK(cl, IndexDomain::entire, te);
+
+  parthenon::par_for(
+      PARTHENON_AUTO_LABEL, 0, pack1.GetNBlocks() - 1, pack1.GetLowerBoundHost(0),
+      pack1.GetUpperBoundHost(0), // This is safe for dense vars
+      kb.s, kb.e, jb.s, jb.e, ib.s, ib.e,
+      KOKKOS_LAMBDA(const int b, const int l, const int k, const int j, const int i) {
+        pack_out(b, l, k, j, i) = w1 * pack1(b, l, k, j, i) + w2 * pack2(b, l, k, j, i);
+      });
+  return TaskStatus::complete;
+}
+
+template <class pack_desc_t>
+void StokesZero(parthenon::CellLevel cl, parthenon::TopologicalElement TeVar,
+                pack_desc_t &pd, MeshData<Real> *out) {
+  auto pack_out = pd.GetPack(out);
+
+  IndexRange ib = out->GetBoundsI(cl, IndexDomain::interior, TeVar);
+  IndexRange jb = out->GetBoundsJ(cl, IndexDomain::interior, TeVar);
+  IndexRange kb = out->GetBoundsK(cl, IndexDomain::interior, TeVar);
+
+  parthenon::par_for(
+      PARTHENON_AUTO_LABEL, 0, pack_out.GetNBlocks() - 1, pack_out.GetLowerBoundHost(0),
+      pack_out.GetUpperBoundHost(0), // This is safe for dense vars only
+      kb.s, kb.e, jb.s, jb.e, ib.s, ib.e,
+      KOKKOS_LAMBDA(const int b, const int l, const int k, const int j, const int i) {
+        pack_out(b, TeVar, l, k, j, i) = 0.0;
+      });
+}
+
+template <class pack_desc_t>
+void StokesComponent(Real fac, parthenon::CellLevel cl,
+                     parthenon::TopologicalElement TeVar,
+                     parthenon::TopologicalElement TeFlux, pack_desc_t &pd, int ndim,
+                     MeshData<Real> *in, MeshData<Real> *out) {
+  auto pack_in = pd.GetPack(in);
+  auto pack_out = pd.GetPack(out);
+
+  IndexRange ib = in->GetBoundsI(cl, IndexDomain::interior, TeVar);
+  IndexRange jb = in->GetBoundsJ(cl, IndexDomain::interior, TeVar);
+  IndexRange kb = in->GetBoundsK(cl, IndexDomain::interior, TeVar);
+  int ioff = TopologicalOffsetI(TeFlux) - TopologicalOffsetI(TeVar);
+  int joff = TopologicalOffsetJ(TeFlux) - TopologicalOffsetJ(TeVar);
+  int koff = TopologicalOffsetK(TeFlux) - TopologicalOffsetK(TeVar);
+  PARTHENON_REQUIRE(ioff == 1 || ioff == 0, "Bad combination of TeVar and TeFlux");
+  PARTHENON_REQUIRE(joff == 1 || joff == 0, "Bad combination of TeVar and TeFlux");
+  PARTHENON_REQUIRE(koff == 1 || koff == 0, "Bad combination of TeVar and TeFlux");
+  PARTHENON_REQUIRE((ioff + joff + koff) == 1, "Bad combination of TeVar and TeFlux");
+  koff = ndim > 2 ? koff : 0;
+  joff = ndim > 1 ? joff : 0;
+
+  parthenon::par_for(
+      PARTHENON_AUTO_LABEL, 0, pack_in.GetNBlocks() - 1, pack_in.GetLowerBoundHost(0),
+      pack_in.GetUpperBoundHost(0), // This is safe for dense vars only
```

Comment:
as above, bounds check in `l` for in and out packs

Reply:
see above, will fix in #1103

Reply:
added bounds check